### PR TITLE
[SPARK-38989][PYTHON] Implement `ignore_index` of `DataFrame/Series.sample`

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Implement `ignore_index` of `DataFrame/Series.sample`

### Why are the changes needed?
To increase pandas API coverage.

### Does this PR introduce _any_ user-facing change?
Yes. `ignore_index` of `DataFrame/Series.sample` is supported as below.
```py
        >>> df = ps.DataFrame({'num_legs': [2, 4, 8, 0],
        ...                    'num_wings': [2, 0, 0, 0],
        ...                    'num_specimen_seen': [10, 2, 1, 8]},
        ...                   index=['falcon', 'dog', 'spider', 'fish'],
        ...                   columns=['num_legs', 'num_wings', 'num_specimen_seen'])
        >>> df 
                num_legs  num_wings  num_specimen_seen
        falcon         2          2                 10
        dog            4          0                  2
        spider         8          0                  1
        fish           0          0                  8

        >>> df.sample(frac=0.5, random_state=1, ignore_index=True)
           num_legs  num_wings  num_specimen_seen
        0         4          0                  2
        1         8          0                  1
        2         0          0                  8
```

### How was this patch tested?
Unit tests.

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
@ueshin @HyukjinKwon @itholic Thank you!

File: python/pyspark/pandas/frame.py [hunk 1]
```diff
@@ -8659,6 +8664,14 @@ def sample(
         falcon         2          2                 10
         fish           0          0                  8
 
+        A random 50% sample of the ``DataFrame``, while ignoring the index.
+
+        >>> df.sample(frac=0.5, random_state=1, ignore_index=True)  # doctest: +SKIP
```
Comment (itholic) [inline python/pyspark/pandas/frame.py:8669]: qq: why do we skip this ?

File: python/pyspark/pandas/frame.py [hunk 2]
```diff
@@ -8689,7 +8702,13 @@ def sample(
         sdf = self._internal.resolved_copy.spark_frame.sample(
             withReplacement=replace, fraction=frac, seed=random_state
         )
-        return DataFrame(self._internal.with_new_sdf(sdf))
+        if ignore_index:
+            spark_index_names = [
+                SPARK_INDEX_NAME_FORMAT(i) for i in range(self._internal.index_level)
+            ]
+            return DataFrame(sdf.drop(*spark_index_names))
```
Comment (itholic) [inline python/pyspark/pandas/frame.py:8709]:
```suggestion
            return DataFrame(sdf.drop(*self._internal.index_spark_column_names))
```

I think maybe `spark_index_names` is already in `self._internal.index_spark_column_names` ?

**Review by itholic — COMMENTED** [review]
Otherwise, looks fine

**Comment by itholic** [comment]
Maybe the JIRA number is incorrect ??

Seems like it should be SPARK-38989.

File: python/pyspark/pandas/frame.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/frame.py:8669]: The sampled rows may not be fixed even with the seed, so we skip the doctests as well as the unit tests.

**Comment by xinrong-databricks** [comment]
Thank you @itholic !

**Review by itholic — APPROVED** [review]
Np! Looks good otherwise.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**